tensorflow/lite/micro/kernels/svdf_test.cc

```diff
@@ -743,81 +703,17 @@ TF_LITE_MICRO_TEST(SvdfQuantized2x2Input2x4OutputShouldMatchGolden) {
-                      // currently known to fail.
-TF_LITE_MICRO_TEST(SvdfFloat1x16Input64x1OutputShouldMatchGolden) {
-  constexpr int batch_size = 1;
-  constexpr int num_units = 64;
-  constexpr int input_size = 16;
-  constexpr int memory_size = 8;
-  constexpr int rank = 1;
-  constexpr int num_filters = num_units * rank;
-  constexpr int activation_state_dims_count =
-      batch_size * memory_size * num_filters;
-  constexpr int output_dims_count = batch_size * num_units;
-  constexpr int input_dims_count = batch_size * input_size;
-
-  float input_data[input_dims_count];
-  float output_data[output_dims_count];
-  float scratch_buffer[batch_size * num_filters];
-  float activation_state_data_mutable[activation_state_dims_count];
-
-  // Initialize activation state to starting values.
-  memcpy(activation_state_data_mutable,
-         tflite::testing::initial_activation_state_data_16x1x1,
-         sizeof(tflite::testing::initial_activation_state_data_16x1x1));
-
-  tflite::testing::TestSVDF(
-      batch_size, num_units, input_size, memory_size, rank, kTfLiteActNone,
-      input_data, tflite::testing::feature_weights_data_16x1x1,
-      tflite::testing::time_weights_data_16x1x1, activation_state_data_mutable,
-      tflite::testing::bias_data_16x1x1, scratch_buffer, output_data,
-      tflite::testing::input_data_16x1x1, input_size,
-      tflite::testing::golden_output_16x1x1);
-}
-
-TF_LITE_MICRO_TEST(SvdfFloat1x16Input64x1OutputReluShouldMatchGolden) {
-  constexpr int batch_size = 1;
-  constexpr int num_units = 64;
-  constexpr int input_size = 16;
-  constexpr int memory_size = 8;
-  constexpr int rank = 1;
-  constexpr int num_filters = num_units * rank;
-  constexpr int activation_state_dims_count =
-      batch_size * memory_size * num_filters;
-  constexpr int output_dims_count = batch_size * num_units;
-  constexpr int input_dims_count = batch_size * input_size;
-
-  float input_data[input_dims_count];
-  float output_data[output_dims_count];
-  float scratch_buffer[batch_size * num_filters];
-  float activation_state_data_mutable[activation_state_dims_count];
-
-  // Initialize activation state to starting values.
-  memcpy(activation_state_data_mutable,
-         tflite::testing::initial_activation_state_data_16x1x1,
-         sizeof(tflite::testing::initial_activation_state_data_16x1x1));
-
-  tflite::testing::TestSVDF(
-      batch_size, num_units, input_size, memory_size, rank, kTfLiteActRelu,
-      input_data, tflite::testing::feature_weights_data_16x1x1,
-      tflite::testing::time_weights_data_16x1x1, activation_state_data_mutable,
-      tflite::testing::bias_data_16x1x1, scratch_buffer, output_data,
-      tflite::testing::input_data_16x1x1, input_size,
-      tflite::testing::golden_output_relu_16x1x1);
-}
-#endif
-
-TF_LITE_MICRO_TEST(SvdfQuantized1x16Input64x1OutputShouldMatchGolden) {
+// Template parameter sets type of both time_weights and activation_state.
+template <typename T>
```

Comment:
I'm not sure these templates are making our tests clearer. I don't have a strong preference, but it could be simpler to just add copies of SvdfQuantized1x16Input64x1OutputShouldMatchGolden and SvdfQuantized1x16Input64x1OutputReluShouldMatchGolden with int8 activation states. I'm curious about Advait's feelings about this.

Reply:
I'm happy to duplicate, especially since we want to remove the int16 version completely once everything is switch over to int8.

Reply:
I changed the templating a bit as Advait suggested below.  Let me know if everyone is okay with this.